packages/manager/modules/iam/src/components/createPolicy/createPolicy.controller.js

```diff
@@ -1,6 +1,226 @@
+   * @type {boolean}
+   */
+  get canSubmit() {
+    return !this.isSubmitting && this.form?.$valid;
+  }
+
+  /**
+   * Whether there is any error coming from the api
+   * @type {boolean}
+   */
+  get hasAnyError() {
+    const errors = Object.values(this.error);
+    const onlySilentErrors = errors.every(({ silent }) => silent);
+    if (!errors.length || onlySilentErrors) {
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Returns a list of form controls with a $validators attribute
+   * E.g. instances of NgModelController
+   * @returns {Object[]}
+   */
+  get formControls() {
+    return (
+      this.form?.$$controls.filter(({ $validators }) => !!$validators) || []
+    );
+  }
+
+  /**
+   * A contextualized Set where each key map to a translation
+   * @returns {Object<string,string>}
+   */
+  get translations() {
+    const pfx = 'iam_create_policy';
+    const { instant: $t } = this.$translate;
+    return {
+      heading: {
+        name: $t(`${pfx}_form_name_heading`),
+        resources: $t(`${pfx}_form_resources_heading`),
+        resourceTypes: $t(`${pfx}_form_resource_types_heading`),
+      },
+      noResources: $t(`${pfx}_error_no_resources`),
+    };
+  }
+
+  $onInit() {
+    // Wait for this.form to be attached
+    this.$timeout(() => {
+      this.formControls.forEach((control) => {
+        const { $name: property } = control;
+        Object.assign(control.$validators, {
+          // Each control is given this validator
+          // It fails if :
+          // - The API returns an error specific to that control
+          // - The model of that control has not changed since the last submission
+          api: (model) =>
+            !this.error[property] ||
+            !isEqual(model, this.modelSnapshot?.[property]),
+        });
+      });
+    });
+  }
+
+  /**
+   * Cancel the policy creation by going back to the previous state
+   * @returns {Promise}
+   */
+  cancelCreation() {
+    return this.goBack();
+  }
+
+  /**
+   * Create the policy
+   * @returns {Promise}
+   */
+  createPolicy() {
+    // >>> TO REMOVE >>>
+    // Needed to demonstrate the current state of this page
+    // Will be removed on next dev iteration
+    const data = {
+      identities: [],
+      permissions: { allow: [{ action: '*' }] },
+      resources: [
+        { urn: 'urn:v1:eu:resource:cdn:28dd67c9-1817-4c02-a3ab-76e96324b603' },
+      ],
+      name: this.model.name,
+    };
+    // <<< TO REMOVE <<<
+
+    this.createSnapshot();
+    this.isSubmitting = true;
+
+    return this.PolicyService.createPolicy(data)
+      .then(() => {
+        this.error = {};
+        return this.goBack({
+          success: {
+            key: 'iam_create_policy_success',
+            values: { name: `<strong>${this.model.name}</strong>` },
+          },
+          reload: true,
+        });
+      })
+      .catch((error) => {
+        this.isSubmitting = false;
```

Comment:
**question:** Shouldn't `this.submitting` set to false in `.then()`. If yes, please move it to `.finally()`

Reply:
This is actually intentional, we keep the spinner displayed on success while the transitions to the previous state (using `goBack`) occurs